### PR TITLE
Check if service is in live mode before sending a broadcast

### PR DESCRIPTION
**Pull request by klssmith**
**Don't create broadcast event unless necessary**
If we're not going to send a broadcast, we don't need to create the
BroadcastEvent in the database. The BroadcastMessage contains all the
data we need - the BroadcastEvent is not used.

Not creating the event when we won't send the broadcast (e.g. when the
broadcast message was created when the service was in trial mode) adds
an extra layer of security.

**Check if a service is live before sending a broadcast**
We only want to send a broadcast if the broadcast message is not stubbed
and the service is live at the point at which the broadcast event should
be created. This is to prevent the situation where a broadcast service is
switched to live / trial mode in between the message being created and
approved (we log an error if this happens).

A stubbed broadcast message with a trial mode service at the point of
approval is not an issue - trial mode services can approve their own
broadcasts. In this situation, we don't create the broadcast event but
also don't need to log an error.

[Pivotal story](https://www.pivotaltracker.com/story/show/177491060)

### CODE REVIEW

File: app/broadcast_message/rest.py
Comment (benthorner) [inline app/broadcast_message/rest.py:198]: Minor: we could invert this and return early (after logging), to avoid some of the nesting. Then it would look like:

```
if stubbed or restricted:
  log_error
  return

rest_of_method
```

**Review by benthorner — APPROVED** [review]
🔒

Just a couple of minor suggestions that you may want to consider before merging.

File: app/broadcast_message/rest.py
Comment (klssmith) [reply app/broadcast_message/rest.py]: 👍 comment updated

File: app/broadcast_message/rest.py
Comment (klssmith) [reply app/broadcast_message/rest.py:198]: We don't want to log an error if the broadcast message is stubbed _and_ the service is restricted (this is the scenario where a trial mode service can approve their own broadcasts) - only if there's a mismatch between the stubbed and restricted values. This means the logic would need to look slightly more complicated if changed:
```python
if stubbed or restricted:
    if stubbed != restricted:
        log_error
  return

rest_of_method
```

Comment (benthorner) [reply app/broadcast_message/rest.py:198]: Oh yes, thanks for explaining. In that case I don't think it makes much difference either way, so let's leave it as-is 👍.

**Pull request merged**
